docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+   - Documentation: [`magic-modules/mmv1/third_party/terraform/website/docs/r`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/r)
+   - Tests: [`magic-modules/mmv1/third_party/terraform/tests`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests)
+   - Sweepers: [`magic-modules/mmv1/third_party/terraform/utils`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/utils)
+4. Modify the Go code as needed.
+   - Replace the comments at the top of the file with the following:
+     ```
+     <% autogen_exception -%>
+     ```
+   - If any of the added Go code (including any imports) is beta-only, change the file suffix to `.go.erb` and wrap the beta-only code in a version guard: `<% unless version = 'ga' -%>...<% else -%>...<% end -%>`.
+5. Register the resource in [`magic-modules/mmv1/third_party/terraform/utils/provider.go.erb`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/provider.go.erb) under "START handwritten resources"
+   - Add a version guard for any beta-only resources.
+{{< /tab >}}
+{{< /tabs >}}
+
+## Add fields
+
+{{< tabs "fields" >}}
+{{< tab "MMv1" >}}
+1. For each API field, copy the following template into the resource's `properties` attribute. Be sure to indent appropriately.
+
+```yaml
+# Supported types: String, Integer, Boolean, Double, Enum,
+# ResourceRef (link to a GCP resource), KeyValuePairs (string -> string map),
+# Array, and NestedObject
+- !ruby/object:Api::Type::String
+  name: 'API_FIELD_NAME'
+  description: |
+    MULTILINE_FIELD_DESCRIPTION
+  # Marks the field (and any subfields) as beta-only. Ensure a beta version block
+  # is present in provider.yaml. Do not use if an ancestor field (or the overall
+  # resource) is already marked as beta-only.
+  # min_version: beta
+
+  # If true, the field (and any subfields) will be considered immutable - that
+  # is, only settable on create. If unset or false, the field will still be
+  # considered immutable if any ancestor field (or the overall resource) is
+  # immutable, unless `update_url` is set.
+  # immutable: true
+
+  # If set, changes to the field's value will trigger a separate call to a
+  # specific API method for updating the field's value. The field will not be
+  # considered immutable even if an ancestor field (or the overall resource) is
+  # immutable.
+  # Terraform field names enclosed in double curly braces will be replaced
+  # with the field values from the resource.
+  # update_url: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}/setFieldName'
+
+  # Overrides the verb used to update this specific field. Allowed values:
```

Comment:
I'm fairly certain this only works with `update_url`.

Reply:
I believe that's correct. Clarified in e7d791daa